### ISSUE TITLE
My runner is green but on execution it stucks on Running with gitlab-runner 10.8.0 (079aad9e)   on iob-gitlab-runner 17ecdb74

### ISSUE THREAD

**Issue by Ferparishuertas** [npalm/terraform-aws-gitlab-runner#15]
Can you help me? Is the spot price to low? No clue...

This is my config file
`concurrent = 10
check_interval = 0

[[runners]]
  name = "iob-gitlab-runner"
  url = "https://gitlab.com/"
  token = "XXXXX"
  executor = "docker+machine"
  limit = 0
  [runners.docker]
    tls_verify = false
    image = "docker:17.11.0-ce"
    privileged = true
    disable_cache = false
    volumes = ["/cache"]
    shm_size = 0
  [runners.cache]
    Type = "s3"
    ServerAddress = "s3-eu-west-1.amazonaws.com"
    AccessKey = "XXXX"
    SecretKey = "XXXXX"
    BucketName = "442729568030-gitlab-runner-cache"
    Insecure = false
  [runners.machine]
    IdleCount = 0
    IdleTime = 600
    MachineDriver = "amazonec2"
    MachineName = "runner-%s"
    MachineOptions = ["amazonec2-access-key=XXXXX", "amazonec2-secret-key=XXX", "amazonec2-instance-type=m4.large", "amazonec2-region=eu-west-1", "amazonec2-vpc-id=vpc-018411147ac4eff0d", "amazonec2-subnet-id=subnet-02b62af9c4fc90770", "amazonec2-private-address-only=true", "amazonec2-request-spot-instance=true", "amazonec2-spot-price=0.04", "amazonec2-security-group=preproduction-docker-machine20180731072557855900000001", "amazonec2-tags=environment,preproduction"]
    OffPeakTimezone = ""
    OffPeakIdleCount = 0
    OffPeakIdleTime = 0`

**Comment by npalm** [comment]
Hi,

I do not think your spot price instance is too low. Once a spot request is created and can't be fulfilled you see that in the tab spot-instances. It looks like you use an older version. Please can you use the master or the latest release tag 1.2.0. You should be able just to start the runner. It will be launched (in the example) in a private network. Once the runner is up you should see a instance. After a while you should have logging in cloudwatch. Do you see the logging. Can take a minute or 5. Is the runner registered in gitlab?

**Comment by npalm** [comment]
Are you using the provided example?

**Comment by Ferparishuertas** [comment]
This is the config. Same as example

#############
# Credentials
#############

provider "aws" {
  access_key = "${var.aws_access_key}"
  secret_key = "${var.aws_secret_key}"
  region     = "${var.region}"
}

data "terraform_remote_state" "iob-dbaas-tf-state" {
  backend = "s3"

  config {
    bucket     = "iobuilders-terraform-state"
    key        = "env://${var.workspace}/XXXX-YYY/terraform.tfstate"
    region     = "${var.region}"
  }
}



module "gitlab-runner" {
  source = "npalm/gitlab-runner/aws"
  version = "1.2.0"

  aws_region              = "${var.region}"
  environment             = "${var.environment}"
  ssh_public_key          = "${file("${var.ssh_key_file}")}"

  vpc_id                  = "${data.terraform_remote_state.iob-dbaas-tf-state.vpc_id}"
  subnet_id_gitlab_runner = "${element(split(",",data.terraform_remote_state.iob-dbaas-tf-state.subnet_ids), 0)}"
  subnet_id_runners       = "${element(split(",",data.terraform_remote_state.iob-dbaas-tf-state.subnet_ids), 0)}"

  runners_name            = "${var.runner_name}"
  runners_gitlab_url      = "${var.gitlab_url}"
  runners_token           = "${var.runner_token}"

  runners_off_peak_timezone   = "Europe/Madrid"
  runners_off_peak_idle_count = 0
  runners_off_peak_idle_time  = 60

  # working 9 to 5 :)
  runners_off_peak_periods = "[\"* * 0-9,17-23 * * mon-fri *\", \"* * * * * sat,sun *\"]"

}

**Comment by npalm** [comment]
Looks like I find the problem. My account as a Role: "AWSServiceRoleForEC2Spot" Which is created by Amazon. I guess it was created the first time I created the runners with unlimitted ec2 access. Last week or so I made the access much more restrictive. So looks like this is the problem. The role has this policy:  AWSEC2SpotServiceRolePolicy

**Comment by npalm** [comment]
I have a fix, on branch 15-service-linked-role-cannot-created. Please can you verify the fix? You can load the fix as follow:

```
source = "github.com/npalm/terraform-aws-gitlab-runner?ref=15-service-linked-role-cannot-created"
```
Thank you for reporting the bug

**Comment by npalm** [comment]
Alternatively you can create the following service linked roles:
```
resource "aws_iam_service_linked_role" "spot" {
  aws_service_name = "spot.amazonaws.com"
}

resource "aws_iam_service_linked_role" "spot" {
  aws_service_name = "autoscaling.amazonaws.com"
}
```

**Comment by npalm** [comment]
Issue is resolved in release 1.3.0

**Comment by Ferparishuertas** [comment]
I tried 1.5.0, spot instances are being created but no job running

Seems docker not installes on spot intances 

75 gitlab-runner[8742]: time="2018-08-24T10:41:33Z" level=info msg="Error getting SSH command to check if the daemon is up: ssh command error:" driver=amazonec2 name=runner-17ecdb74-runner-1535106043-bd5f07e0 ope
ration=create
Aug 24 10:41:33 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:33Z" level=info msg="command : sudo docker version" driver=amazonec2 name=runner-17ecdb74-runner-1535106043-bd5f07e0 operation=create
Aug 24 10:41:33 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:33Z" level=info msg="err     : exit status 1" driver=amazonec2 name=runner-17ecdb74-runner-1535106043-bd5f07e0 operation=create
Aug 24 10:41:33 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:33Z" level=info msg="output  : sudo: docker: command not found" driver=amazonec2 name=runner-17ecdb74-runner-1535106043-bd5f07e0 operation=create
Aug 24 10:41:34 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:34Z" level=info msg="Error getting SSH command to check if the daemon is up: ssh command error:" driver=amazonec2 name=runner-17ecdb74-runner-1535106043-bcbc362c ope
ration=create
Aug 24 10:41:34 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:34Z" level=info msg="command : sudo docker version" driver=amazonec2 name=runner-17ecdb74-runner-1535106043-bcbc362c operation=create
Aug 24 10:41:34 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:34Z" level=info msg="err     : exit status 1" driver=amazonec2 name=runner-17ecdb74-runner-1535106043-bcbc362c operation=create
Aug 24 10:41:34 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:34Z" level=info msg="output  : sudo: docker: command not found" driver=amazonec2 name=runner-17ecdb74-runner-1535106043-bcbc362c operation=create
Aug 24 10:41:36 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:36Z" level=info msg="Error getting SSH command to check if the daemon is up: ssh command error:" driver=amazonec2 name=runner-17ecdb74-runner-1535106043-bd5f07e0 ope
ration=create
Aug 24 10:41:36 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:36Z" level=info msg="command : sudo docker version" driver=amazonec2 name=runner-17ecdb74-runner-1535106043-bd5f07e0 operation=create
Aug 24 10:41:36 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:36Z" level=info msg="err     : exit status 1" driver=amazonec2 name=runner-17ecdb74-runner-1535106043-bd5f07e0 operation=create
Aug 24 10:41:36 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:36Z" level=info msg="output  : sudo: docker: command not found" driver=amazonec2 name=runner-17ecdb74-runner-1535106043-bd5f07e0 operation=create
Aug 24 10:41:37 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:37Z" level=info msg="Error getting SSH command to check if the daemon is up: ssh command error:" driver=amazonec2 name=runner-17ecdb74-runner-1535106043-bcbc362c ope
ration=create
Aug 24 10:41:37 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:37Z" level=info msg="command : sudo docker version" driver=amazonec2 name=runner-17ecdb74-runner-1535106043-bcbc362c operation=create
Aug 24 10:41:37 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:37Z" level=info msg="err     : exit status 1" driver=amazonec2 name=runner-17ecdb74-runner-1535106043-bcbc362c operation=create
Aug 24 10:41:37 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:37Z" level=info msg="output  : sudo: docker: command not found" driver=amazonec2 name=runner-17ecdb74-runner-1535106043-bcbc362c operation=create
Aug 24 10:41:40 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:40Z" level=info msg="Error getting SSH command to check if the daemon is up: ssh command error:" driver=amazonec2 name=runner-17ecdb74-runner-1535106043-bcbc362c ope
ration=create
Aug 24 10:41:40 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:40Z" level=info msg="command : sudo docker version" driver=amazonec2 name=runner-17ecdb74-runner-1535106043-bcbc362c operation=create
Aug 24 10:41:40 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:40Z" level=info msg="err     : exit status 1" driver=amazonec2 name=runner-17ecdb74-runner-1535106043-bcbc362c operation=create
Aug 24 10:41:40 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:40Z" level=info msg="output  : sudo: docker: command not found" driver=amazonec2 name=runner-17ecdb74-runner-1535106043-bcbc362c operation=create
Aug 24 10:41:41 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:41Z" level=error msg="Error creating machine: Error running provisioning: Maximum number of retries (60) exceeded" driver=amazonec2 name=runner-17ecdb74-runner-15351
06043-bd5f07e0 operation=create
Aug 24 10:41:41 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:41Z" level=info msg="The default lines below are for a sh/bash shell, you can specify the shell you're using, with the --shell flag." driver=amazonec2 name=runner-17
ecdb74-runner-1535106043-bd5f07e0 operation=create
Aug 24 10:41:42 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:42Z" level=warning msg="Machine creation failed, trying to provision" error="exit status 1" name=runner-17ecdb74-runner-1535106043-bd5f07e0
Aug 24 10:41:43 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:43Z" level=info msg="Waiting for SSH to be available..." name=runner-17ecdb74-runner-1535106043-bd5f07e0 operation=provision
Aug 24 10:41:43 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:43Z" level=info msg="Detecting the provisioner..." name=runner-17ecdb74-runner-1535106043-bd5f07e0 operation=provision
Aug 24 10:41:45 ip-10-0-1-75 gitlab-runner[8742]: time="2018-08-24T10:41:45Z" level=error msg="Error creating machine: Error running provisioning: Maximum number of retries (60) exceeded" driver=amazonec2 name=runner-17ecdb74-runner-15351
06043-bcbc362c operation=create
